social_core/backends/base.py

```diff
@@ -116,6 +117,8 @@ def extra_data(self, user, uid, response, details=None, *args, **kwargs):
             # store the last time authentication toke place
             'auth_time': int(time.time())
         }
+        if self.GET_ALL_EXTRA_DATA or self.setting('GET_ALL_EXTRA_DATA', False):
+            self.EXTRA_DATA = [data_key for data_key in response]
```

Comment:
Don't modify the `self.EXTRA_DATA` directly, assign it to a var, something like:

```
extra_data_keys = []
if self.GET_ALL_EXTRA_DATA or self.setting('GET_ALL_EXTRA_DATA', False):
    extra_data_entries = response.keys()
else:
    extra_data_entries = (self.EXTRA_DATA or []) + self.setting('EXTRA_DATA', [])

for entry in extra_data_entries:
  ...
```